domain/schema/controller.go

```diff
@@ -113,7 +113,20 @@ CREATE TABLE change_log (
     CONSTRAINT          fk_change_log_namespace
             FOREIGN KEY (namespace_id)
             REFERENCES  change_log_namespace(id)
-);`[1:]
+);
+
+-- The change log witness table is used to track which nodes have seen
+-- which change log entries. This is used to determine when a change log entry
+-- can be deleted.
+-- We'll delete all change log entries that are older than the oldest change
+-- log entry that has been seen by all controllers.
+CREATE TABLE change_log_witness (
+    tag                 TEXT PRIMARY KEY,
```

Comment:
We will need to establish some things here. 

I probably need to finish my patch for recording controller nodes, at which point this would be the UUID of the controller, so `tag` should probably be names something else.

`last_seen_at` is not a great name, because it's the time that we recorded the watermark, not when we dispatched the term that the `change_log_id` was in.

`last_deleted_at` needs some thought. Are all controllers going to be deleting from the change log? If so, does the column get updated if nothing is deleted by a particular controller?